src/main/java/ua/bookstore/online/service/impl/CartItemServiceImpl.java

```diff
@@ -0,0 +1,65 @@
+package ua.bookstore.online.service.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import ua.bookstore.online.dto.shopping.cart.CartItemRequestDto;
+import ua.bookstore.online.dto.shopping.cart.CartItemResponseDto;
+import ua.bookstore.online.dto.shopping.cart.QuantityDto;
+import ua.bookstore.online.exception.EntityNotFoundException;
+import ua.bookstore.online.mapper.CartItemMapper;
+import ua.bookstore.online.model.Book;
+import ua.bookstore.online.model.CartItem;
+import ua.bookstore.online.model.ShoppingCart;
+import ua.bookstore.online.repository.book.BookRepository;
+import ua.bookstore.online.repository.shopping.cart.item.CartItemRepository;
+import ua.bookstore.online.service.CartItemService;
+
+@Service
+@RequiredArgsConstructor
+public class CartItemServiceImpl implements CartItemService {
+    public static final int ONE_MORE = 1;
+    private final BookRepository bookRepository;
+    private final CartItemRepository cartItemRepository;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public CartItemResponseDto add(CartItemRequestDto requestDto, ShoppingCart shoppingCart) {
+        Book book = bookRepository.findById(requestDto.bookId()).orElseThrow(
+                () -> new EntityNotFoundException("Book not found by id " + requestDto.bookId()));
+
+        CartItem cartItem = cartItemRepository.findByBookAndShoppingCart(book, shoppingCart)
+                                               .map(item -> {
+                                                   item.setQuantity(item.getQuantity() + ONE_MORE);
+                                                   return item;
```

Comment:
Maybe I misunderstood, but if you already have a book in your cart, you add 1 to the quantity? And why not the quantity that the customer provides in the request?

Reply:
Because I can change the quantity by the endpoint that does it.
But maybe you are right.